source/validate_instruction.cpp

```diff
@@ -131,6 +131,68 @@ spv_result_t CapCheck(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Checks that the number of characters in a string doesn't exceed the limit.
+spv_result_t LimitCheckString(ValidationState_t& _,
+                              const spv_parsed_instruction_t* inst) {
+  // By definition, num_words of an operand is a 16-bit unsigned which will not
+  // exceed the limit of characters for a string literal.
+  // This function checks that a NULL termination must be found before reaching
+  // the end of the string words.
+  for (int i = 0; i < inst->num_operands; ++i) {
+    auto operand = inst->operands[i];
+    if (SPV_OPERAND_TYPE_LITERAL_STRING == operand.type) {
+      bool null_terminated = false;
+      for (auto word_i = 0; !null_terminated && word_i < operand.num_words;
+           ++word_i) {
+        const uint32_t word = inst->words[operand.offset + word_i];
+        // check if any BYTE in the word is NULL.
+        if ((word & 0x000000ff) == 0 || (word & 0x0000ff00) == 0 ||
+            (word & 0x00ff0000) == 0 || (word & 0xff000000) == 0) {
+          null_terminated = true;
+        }
+      }
+      if (!null_terminated) {
+        return _.diag(SPV_ERROR_INVALID_BINARY)
+               << "Found a string that is not null-terminated.";
```

Comment:
Need a test case to trigger this.
I recommend creating test strings programmatically.  :-)